File: generated-api-playwright/utils/apiAssertions.ts
import { expect, APIResponse } from '@playwright/test';

export async function expectHealthyStatus(response: APIResponse, expectedStatus: number, operation: string) {
  const status = response.status();
  if ([401, 403].includes(status)) throw new Error(`[RCA:API_AUTHORIZATION] ${operation} returned ${status}. Check token, role, session, VPN/VDI or environment.`);
  if (status >= 500) throw new Error(`[RCA:API_SERVER_OR_ENVIRONMENT] ${operation} returned ${status}. Do not self-heal tests until backend/environment is checked.`);
  expect(status, `${operation}: HTTP status`).toBe(expectedStatus);
}

export async function expectJsonIfJson(response: APIResponse, operation: string) {
  const contentType = response.headers()['content-type'] || '';
  if (contentType.includes('application/json')) {
    const body = await response.json();
    expect(body, `${operation}: JSON body`).toBeTruthy();
    return body;
  }
  return null;
}

File: generated-api-restassured/src/test/java/com/aiqa/api/support/ApiAssertions.java
package com.aiqa.api.support;

import io.restassured.response.Response;
import static org.junit.jupiter.api.Assertions.*;

public class ApiAssertions {
  public static void expectHealthyStatus(Response response, int expected, String operation) {
    int status = response.statusCode();
    if (status == 401 || status == 403) fail("[RCA:API_AUTHORIZATION] " + operation + " returned " + status + ". Check token, role, session, VPN/VDI or environment.");
    if (status >= 500) fail("[RCA:API_SERVER_OR_ENVIRONMENT] " + operation + " returned " + status + ". Do not self-heal tests until backend/environment is checked.");
    assertEquals(expected, status, operation + " HTTP status");
  }
}
